gbpservice/nfp/orchestrator/modules/service_orchestrator.py

```diff
@@ -1477,24 +1549,44 @@ def get_port_info(self, port_id):
             return None
 
     def get_network_function_details(self, network_function_id):
-        network_function = self.db_handler.get_network_function(
-            self.db_session, network_function_id)
+        network_function = None
+        network_function_instance = None
+        network_function_device = None
+        service_type = None
+
+        nfp_context = nfp_core_context.get_nfp_context()
+        if nfp_context:
+            network_function = nfp_context['network_function']
+            network_function_instance = nfp_context['network_function_instance']
+            network_function_device = nfp_context['network_function_device']
+            service_type = nfp_context['service_details']['service_type']
+
+        if not network_function:
+            network_function = self.db_handler.get_network_function(
```

Comment:
In what cases network function wont be present, should we raise error if network function is not present ?